### PR TITLE
Step 822 pathutil update

### PR DESCRIPTION
**Pull request by godrei**
Move `FilterPaths` and `SortPathsByComponents` functionality from `bitrise-init/utility` to `go-utils/pathutil`, in order to make it independent from bitrise-init and make it more visible.

### CODE REVIEW

File: pathutil/path_filter.go [hunk 1]
```diff
@@ -97,20 +98,20 @@ func IsDirectoryFilter(allowed bool) FilterFunc {
...
-		in := filepath.Dir(pth) == dir
-		return allowed == in, nil
+		in := (filepath.Dir(pth) == dir)
+		return (allowed == in), nil
 	}
 }
 
-// DirectoryContainsFile returns a FilterFunc that checks if a directory contains a file
-func DirectoryContainsFile(fileName string) FilterFunc {
+// DirectoryContainsFileFilter returns a FilterFunc that checks if a directory contains a file
```
Comment (mateherber) [inline pathutil/path_filter.go:113]: Is this rename intentional?

File: pathutil/path_filter_test.go [hunk 2]
```diff
@@ -1,15 +1,16 @@
-package fsfilter
+package pathutil_test
```
Comment (mateherber) [inline pathutil/path_filter_test.go:1]: package needs to be pathutil_test?

**Review by mateherber — CHANGES_REQUESTED** [review]

File: pathutil/path_filter.go (hunk 1 above)
Comment (godrei) [reply pathutil/path_filter.go:113]: All the other FilterFunc generators are `XYZFilter`s

Comment (mateherber) [reply pathutil/path_filter.go:113]: 👌🏻

File: pathutil/path_filter_test.go (hunk 2 above)
Comment (godrei) [reply pathutil/path_filter_test.go:1]: In order to overcome pathutil - fileutil reference circle. 
The test requires fileutil, but fileutil includes pathutil.

Comment (mateherber) [reply pathutil/path_filter_test.go:1]: 😢

**Review by mateherber — APPROVED** [review]
